### PR TITLE
Enhancements to parallelism of query encryption.

### PR DESCRIPTION
**Pull request by tellison**
- Switch parallel encrypt task to use callable interface.
 - Use parallelStream to generate the (in-memory) exponent table.
 - Run some test material through the parallel path.

### CODE REVIEW

**Comment by tellison** [comment]
Not tested on distributed.

File: src/main/java/org/apache/pirk/query/wideskies/Query.java [hunk 1]
```diff
@@ -136,89 +124,30 @@ public boolean containsElement(BigInteger element)
...
-    queryElements.clear();
-  }
-
   /**
    * This should be called after all query elements have been added in order to generate the expTable. For int exponentiation with BigIntegers, assumes that
    * dataPartitionBitSize < 32.
-   *
    */
-  public void generateExpTable(int numThreads) throws InterruptedException
+  public void generateExpTable()
```
Comment (tellison) [inline src/main/java/org/apache/pirk/query/wideskies/Query.java]: I want to draw attention to this part of the new code, that will use the ```ForkJoinPool#commonPool()``` (i.e. all available processors by default).  We can put this into a smaller custom pool if it is desirable to have that restriction.

Comment (ellisonanne) [inline src/main/java/org/apache/pirk/query/wideskies/Query.java]: Allowing this computation to use all possible resources makes sense here...

**Comment by ellisonanne** [comment]
+1 - nice streamlining

I will run the distributed tests before merging - if merged and the PR closed, please assume that the tests passed. If not, I will report back accordingly.

**Pull request closed without merging**